Cinnamon Survey 20220325

Cyanoxygen opened the pull request:
Topic Description
-----------------
- Update cinnamon DE to the latest version
- Revert `caribou` package drop - cinnamon can not launch without this

Package(s) Affected
-------------------

- `groups/cinnamon`
- `caribou`

Security Update?
----------------

No

Build Order
-----------

- `caribou`
- `groups/cinnamon`

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Cyanoxygen committed:
Revert "caribou: drop, orphaned"

This reverts commit 582d54611040400d2b355929a4785d8217fe7e5e.

Cyanoxygen committed:
cinnamon-desktop: update to 5.2.1

Cyanoxygen committed:
warpinator: update to 1.2.9

Cyanoxygen committed:
cinnamon-menus: update to 5.2.0

Cyanoxygen committed:
bulky: update to 2.1

Cyanoxygen committed:
cinnamon-control-center: update to 5.2.1

Cyanoxygen committed:
cjs: update to 5.2.0

Cyanoxygen committed:
cinnamon-settings-daemon: update to 5.2.0

Cyanoxygen committed:
muffin: update to 5.2.1

Cyanoxygen committed:
cinnamon-translations: update to 5.2.2

Cyanoxygen committed:
cinnamon: update to 5.2.7

Cyanoxygen committed:
sticky: update to 1.8

Cyanoxygen committed:
nemo: update to 5.2.4

Cyanoxygen committed:
cinnamon-session: update to 5.2.0

Cyanoxygen committed:
cinnamon-screensaver: update to 5.2.1

Cyanoxygen committed:
cinnamon: no need to move Python modules now

Cyanoxygen committed:
cinnamon-desktop: rebuild patch

Cyanoxygen committed:
caribou: update to 0.4.21+git20210310

Cyanoxygen committed:
caribou: fix build errors
- force Python 3
- use manual build script

Cyanoxygen committed:
cinnamon: add requests as dependency

to fix the broken cinnamon-settings

Cyanoxygen commented:
Testing
------

The updated Cinnamon DE is running well on my test machine (a Virtualbox VM):
![image](https://user-images.githubusercontent.com/29174191/160115326-a584d021-1592-49f9-ac25-d085079ba21d.png)

WhiredPlanck commented on extra-gnome/caribou/autobuild/defines:
```diff
@@ -0,0 +1,9 @@
+BUILDDEP="gtk-2 intltool docbook-xsl"
+
+
+NOLTO=1
```
Why?

Cyanoxygen replied:
Oh, probably was left over while I experimenting building this package.

Cyanoxygen committed:
caribou: remove unused and duplicate build options

Cyanoxygen committed:
caribou: add gnome-common as build dependency

Cyanoxygen committed:
caribou: add vala as dependency

MingcongBai commented on extra-gnome/caribou/autobuild/build:
```diff
@@ -0,0 +1,19 @@
+abinfo "Running autogen.sh..."
+NOCONFIGURE=1 $SRCDIR/autogen.sh
+
+abinfo "Unset PYTHON to disable Python 2"
```
You could use “PYTHON=“ in AUTOTOOLS_AFTER to achieve the same effect.

Cyanoxygen committed:
caribou: revert to version 0.4.21, use git tag instead

- `configure.ac` is needed to force usage of Python 3 as Gi is no longer
  available to Python 2 and results in a build error, and such file only
  exists in Git tags.

- Patching configure script is not considered as this script is generated.

Cyanoxygen committed:
caribou: use autobuild3 to handle autotools build system

Cyanoxygen committed:
caribou: [Upstream patch] backport patches to fix build
